src/text/raw_text_writer.rs

```diff
@@ -132,15 +139,55 @@ struct EncodingLevel {
     child_count: usize,
 }
 
+struct WhitespaceConfig {
+    // Top-level values are independent of other values in the stream, we may separate differently
+    space_between_top_level_values: &'static str,
+    // Non-top-level values are within a container
+    space_between_nested_values: &'static str,
+    indentation: &'static str,
+    space_after_field_name: &'static str,
+    space_after_container_start: &'static str,
+}
+
+static DEFAULT_WS_CONFIG: WhitespaceConfig = WhitespaceConfig {
```

Comment:
It's subjective, but I have a preference for spelling variable names out. Could we make these `WS`s into `WHITESPACE`s?

Reply:
Done!